sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClientBuilder.java

```diff
@@ -195,13 +195,26 @@ public ConfigurationClientBuilder endpoint(String endpoint) {
      * Sets the credential to use when authenticating HTTP requests. Also, sets the {@link #endpoint(String) endpoint}
      * for this ConfigurationClientBuilder.
      *
-     * @param credential The credential to use for authenticating HTTP requests.
+     * @param connectionString Connection string in the format "endpoint={endpoint_value};id={id_value};
+     * secret={secret_value}"
      * @return The updated ConfigurationClientBuilder object.
      * @throws NullPointerException If {@code credential} is {@code null}.
      */
-    public ConfigurationClientBuilder credential(ConfigurationClientCredentials credential) {
-        this.credential = Objects.requireNonNull(credential);
+    public ConfigurationClientBuilder connectionString(String connectionString) {
+        Objects.requireNonNull(connectionString);
+
+        try {
+            this.credential = new ConfigurationClientCredentials(connectionString);
+        } catch (InvalidKeyException err) {
+            throw logger.logExceptionAsError(new IllegalArgumentException(
```

Comment:
Pass the err as the inner parameter to IllegalArgumentException. Same with the one below so users can expand the inner throwable.